cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -416,11 +400,11 @@ void reader::impl::decode_page_data(read_mode mode, size_t skip_rows, size_t num
         final_offsets.emplace_back(offset);
         out_buf.user_data |= PARQUET_COLUMN_BUFFER_FLAG_LIST_TERMINATED;
       } else if (out_buf.type.id() == type_id::STRING) {
-        // need to cap off the string offsets column
-        auto const sz = static_cast<size_type>(col_string_sizes[idx]);
-        if (sz <= strings::detail::get_offset64_threshold()) {
+        // only if it is not a large strings column
+        if (col_string_sizes[idx] <=
+            static_cast<size_t>(strings::detail::get_offset64_threshold())) {
```

Comment:
`strings::detail::get_offset64_threshold()` is always `<= std::numeric_limits<int32_t>::max()`

Reply:
That is true. The purpose is for testing (and requested by Spark) as way you could test with a lower threshold without needing to create a 2GB char array.

Reply:
Yep,I was trying to comment on why I changed to upcasting `threshold` instead of downcasting `col_string_sizes[idx]` as `col_string_sizes` may be > 2B for large str cols.